### PR TITLE
split apart get api function

### LINKED ISSUE #140: split apart getApi()
As a developer, we want to pull apart getApi() to be able to make it more readable and allow for different uses of the functionality.

AC:
- New functions from getAPI:
  - disconnectApi()
  - createAndConnectToApi()
  - createApi()
- Update how we handle subscriptions where possible in the function.

### PR DESCRIPTION
**Pull request by claireolmstead**
closes #140 
~~blocked by #139 (only affects Provider and Capacity components)~~

### CODE REVIEW

**Review by rlaferla — COMMENTED** [review]
Provider and Capacity blocks show "Not Connected"

https://github.com/LibertyDSNP/provider-dashboard/assets/1135694/90f12991-dec2-42bf-a051-ad40f9d6b094

File: src/lib/polkadotApi.ts [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
+async function updateChainInfo(api: ApiPromise) {
+  const chain = await api.rpc.system.properties();
+  const token = getToken(chain);
+  const blockNumber = (await getBlockNumber(api)) as bigint;
+  const epochNumber = await getEpoch(api);
+  storeChainInfo.set({ connected: true, blockNumber, epochNumber, token });
+}
+
+export async function createAndConnectToApi(networkEndpoint: string) {
+  dotApi.subscribe(async (apiState: DotApi) => await apiState.api?.disconnect());
```
Comment (wilwade) [inline src/lib/polkadotApi.ts:23]: I don't understand this line. When the dotApi store changes we trigger a disconnect?

**Comment by claireolmstead** [comment]
> Provider and Capacity blocks show "Not Connected"
> 
>  Screen.Recording.2024-02-06.at.9.05.45.AM.mov

In the top comment I say "blocked by https://github.com/LibertyDSNP/provider-dashboard/pull/139 (only affects Provider and Capacity components)".  #139 will be merged in first and this will not happen.

File: src/lib/polkadotApi.ts (hunk 1 above)
Comment (claireolmstead) [reply src/lib/polkadotApi.ts:23]: @wilwade The way I understood it, we want to disconnect the old API instance before creating a new instance. Shannon's code worked the same way.

Comment (claireolmstead) [reply src/lib/polkadotApi.ts]: Ended up deleting. It was causing problems.

File: src/components/Provider.svelte [hunk 2]
```diff
@@ -6,11 +6,12 @@
...
   import ListCard from './ListCard.svelte';
   import { ActionForms } from '$lib/storeTypes.js';
-  import { afterUpdate } from 'svelte';
+  import { onMount } from 'svelte';
 
   let accountBalances: AccountBalances = { free: 0n, reserved: 0n, frozen: 0n };
 
-  afterUpdate(async () => {
+  onMount(async () => {
+    console.log($dotApi.selectedEndpoint);
```
Comment (enddynayn) [inline src/components/Provider.svelte:14]: left over.

File: src/components/ProviderLogin.svelte [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
   $: canConnect = $user?.network != null && $providerAccountsStore.size > 0 && $user?.address !== '';
 
-  function connect() {
+  async function connect() {
+    if (!$user.network?.endpoint?.origin) {
+      alert('Error connecting to endpoint.');
+      return;
+    }
+    console.log('HERE USER', $user);
+    await createAndConnectToApi($user.network?.endpoint?.origin);
```
Comment (enddynayn) [inline src/components/ProviderLogin.svelte:22]: this requires a string but this can be undefined?

**Review by mattheworris — APPROVED** [review]
🚢 it!

- Read through the code changes.
- Executed the app and clicked through some test cases.

File: src/components/ProviderLogin.svelte [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
   let canConnect: boolean = false;
   $: canConnect = $user?.network != null && $providerAccountsStore.size > 0 && $user?.address !== '';
 
-  function connect() {
+  async function connect() {
+    if (!$user.network?.endpoint?.origin) {
+      alert('Error connecting to endpoint.');
+      return;
+    }
+    console.log('HERE USER', $user);
```
[the remark this answers is not in the document]
Comment (claireolmstead) [reply src/components/ProviderLogin.svelte]: opted to delete this.

File: test/unit-and-integration/capacity.test.ts [hunk 5]
```diff
@@ -2,7 +2,6 @@ import { cleanup, render, waitFor } from '@testing-library/svelte';
 import '@testing-library/jest-dom';
 import { storeChainInfo, dotApi } from '../../src/lib/stores';
 import Capacity from '$components/Capacity.svelte';
```
[the remark this answers is not in the document]
Comment (claireolmstead) [reply test/unit-and-integration/capacity.test.ts:4]: added back in.

**Pull request merged**